sympy/tensor/array/arrayop.py

```diff
@@ -130,9 +130,16 @@ def tensorcontraction(array, *contraction_axes):
     for icontrib in itertools.product(*remaining_indices):
         i = sum(icontrib)
         isum = S.Zero
+        summed_deltas = []
         for axes_group in contraction_axes:
+            lidx = []
             for js in range(array.shape[axes_group[0]]):
-                isum += array[i + sum([cum_shape[ig]*js for ig in axes_group])]
+                lidx.append(sum([cum_shape[ig]*js for ig in axes_group]))
+            summed_deltas.append(lidx)
+
+        for sum_to_index in itertools.product(*summed_deltas):
+            isum += array[i + sum(sum_to_index)]
```

Comment:
I think it's better to remane the variable _i_ because it is already used in a previous for-loop.